### PR TITLE
Compile MTI feature files that are embedded in UFOs

### PR DESCRIPTION
**Pull request by brawer**
https://github.com/googlei18n/fontmake/issues/289

### CODE REVIEW

File: Lib/ufo2ft/__init__.py [hunk 1]
```diff
@@ -52,17 +55,40 @@ def compileTTF(ufo, outlineCompilerClass=OutlineTTFCompiler,
...
 
     return otf
+
+
+def _getMtiFeatures(ufo, mtiFeaFiles=None):
+    features = {}
+    prefix = "com.github.googlei18n.ufo2ft.mtiFeatures/"
+    for fileName in ufo.data.fileNames:
+        if fileName.startswith(prefix) and fileName.endswith(".mti"):
+            features[fileName[len(prefix):-4]] = ufo.data[fileName]
```
Comment (anthrotype) [inline Lib/ufo2ft/__init__.py]: should we validate that the key is one of "GDEF", "GPOS" or "GSUB"?

File: Lib/ufo2ft/__init__.py [hunk 2]
```diff
@@ -52,17 +55,40 @@ def compileTTF(ufo, outlineCompilerClass=OutlineTTFCompiler,
...
+
+    # TODO: Remove the ability to pass MTI feature files as side input
+    # to the compiler. https://github.com/googlei18n/fontmake/issues/289
+    if mtiFeaFiles is not None:
+        log.warn("Future versions of this font compiler will remove "
+                 "the capability to inject MTI features as side input. "
+                 "Instead, please embed your MTI feature files into the "
+                 "comiled UFO sources. See "
```
Comment (adrientetar) [inline Lib/ufo2ft/__init__.py]: typo: compiled

File: Lib/ufo2ft/__init__.py (hunk 1 above)
Comment (brawer) [inline Lib/ufo2ft/__init__.py]: mtiLib will complain when it cannot compile a table. (Also “cmap”, by the way).

File: Lib/ufo2ft/__init__.py [hunk 3]
```diff
@@ -88,7 +88,7 @@ def _getMtiFeatures(ufo, mtiFeaFiles=None):
         # For now, the side input wins over any MTI features inside the UFO.
         features = {}
         for table, feapath in mtiFeaFiles.items():
-            with open(feapath, "r") as feafile:
+            with open(feapath, "rb") as feafile:
```
Comment (anthrotype) [inline Lib/ufo2ft/__init__.py]: are you sure mtiLib wants bytes? It seems to me it opens the files as text and decodes them as UTF-8:
https://github.com/fonttools/fonttools/blob/4e9806a/Lib/fontTools/mtiLib/__init__.py#L1178

File: Lib/ufo2ft/__init__.py (hunk 2 above)
Comment (brawer) [inline Lib/ufo2ft/__init__.py]: Fixed. Thanks for noticing.

File: Lib/ufo2ft/__init__.py (hunk 1 above)
Comment (anthrotype) [inline Lib/ufo2ft/__init__.py]: ah, ok fine

File: Lib/ufo2ft/__init__.py [hunk 4]
```diff
@@ -52,17 +55,40 @@ def compileTTF(ufo, outlineCompilerClass=OutlineTTFCompiler,
...
+    features = {}
+    prefix = "com.github.googlei18n.ufo2ft.mtiFeatures/"
+    for fileName in ufo.data.fileNames:
+        if fileName.startswith(prefix) and fileName.endswith(".mti"):
+            features[fileName[len(prefix):-4]] = ufo.data[fileName]
+
+    # TODO: Remove the ability to pass MTI feature files as side input
+    # to the compiler. https://github.com/googlei18n/fontmake/issues/289
+    if mtiFeaFiles is not None:
+        log.warn("Future versions of this font compiler will remove "
```
Comment (anthrotype) [inline Lib/ufo2ft/__init__.py]: for me, we could just drop support for this straight away. I think Noto is the only user of this.

File: Lib/ufo2ft/__init__.py (hunk 3 above)
Comment (brawer) [inline Lib/ufo2ft/__init__.py]: Fixed (I believe).

File: Lib/ufo2ft/__init__.py (hunk 4 above)
Comment (brawer) [inline Lib/ufo2ft/__init__.py]: OK to leave this in for 1 week? Otherwise it’ll be tricker to coordinate how we update the dependencies of fontmake.

Comment (anthrotype) [inline Lib/ufo2ft/__init__.py]: sure, no problem

**Pull request merged**